recipes/llvm-core/all/conanfile.py

```diff
@@ -179,33 +180,24 @@ def package(self):
         cmake.install()
 
         if not self.options.shared:
-            lib_regex = re.compile(
-                r'add_library\((\w+?)\s.*?\)'
-                r'(?:(?:#|\w|\s|\()+?'
-                r'INTERFACE_LINK_LIBRARIES\s\"((?:;|:|/|\.|\w|\s|-|\(|\))+?)\"'
-                r'(?:.|\n)*?\))?'
-            )
-            exports_file = 'LLVMExports.cmake'
-            exports_path = package_path.joinpath('lib', 'cmake', 'llvm')
-            exports_path = exports_path.joinpath(exports_file)
-
-            exports = tools.load(str(exports_path.resolve()))
-            exports = exports.replace(r'\$<LINK_ONLY:', '')
-            exports = exports.replace(r'>', '')
-
-            components = {}
-            for match in re.finditer(lib_regex, exports):
-                lib, deps = match.groups()
+            self.run('cmake --graphviz=graph/llvm.dot .')
+            with tools.chdir('graph'):
+                dot_text = tools.load('llvm.dot').replace(' ', '')
+
+            dep_regex = re.compile(r'//(.+)->(.+)$', re.MULTILINE)
+            deps = re.findall(dep_regex, dot_text)
```

Comment:
What do you think about this @intelligide? Instead of parsing `LLVMExports.cmake`, I use **cmake** to output the dependency graph as a **graphviz**'s *dot* file which is way easier to parse.

It still uses a regular expression, but it is a simple one. The alternative would be using external tools like **graphviz** plus **networkx** that will do the parsing for you, but then you'll introduce a dependency on those tools that may not be installed along **conan**. I think this solution is easy and maintainable enough.

Reply:
It's better (not perfect yet but better than nothing).